internal/service/handlers/authorize.go

```diff
@@ -35,59 +39,91 @@ func Authorize(w http.ResponseWriter, r *http.Request) {
 			Log(r).WithError(err).Info("Failed to verify proof")
 			ape.RenderErr(w, problems.Unauthorized())
 			return
 		}
 	}
 
-	access, aexp, err := JWT(r).IssueJWT(
-		&jwt.AuthClaim{
-			Nullifier: req.Data.ID,
-			Type:      jwt.AccessTokenType,
-		},
-	)
+	verified := Points(r).DefaultVerified
 
+	if !Points(r).Disabled {
+		verified, err = findOutVerificationStatus(r, req)
+		if err != nil {
+			Log(r).WithError(err).Warnf("failed to find out passport verification status for [%s]; IsVerified set to false", req.Data.ID)
+		}
+	}
+
+	access, refresh, aexp, rexp, err := issueJWTs(r, req.Data.ID, verified)
 	if err != nil {
-		Log(r).WithError(err).WithField("user", req.Data.ID).Error("failed to issuer JWT token")
+		Log(r).WithError(err).WithField("user", req.Data.ID).Error("failed to issue JWTs")
 		ape.RenderErr(w, problems.InternalError())
 		return
 	}
 
-	refresh, rexp, err := JWT(r).IssueJWT(
-		&jwt.AuthClaim{
-			Nullifier: req.Data.ID,
-			Type:      jwt.RefreshTokenType,
-		},
-	)
+	Cookies(r).SetAccessToken(w, access, aexp)
+	Cookies(r).SetRefreshToken(w, refresh, rexp)
+	ape.Render(w, newTokenResponse(req.Data.ID, access, refresh))
+}
 
+func findOutVerificationStatus(r *http.Request, req *resources.AuthorizeRequest) (bool, error) {
+	reqV, err := http.NewRequest("GET", Points(r).URL+Points(r).Endpoint+req.Data.ID, nil)
 	if err != nil {
-		Log(r).WithError(err).WithField("user", req.Data.ID).Error("failed to issuer JWT token")
-		ape.RenderErr(w, problems.InternalError())
-		return
+		return false, fmt.Errorf("failed to create GET request: %w", err)
 	}
 
-	resp := resources.TokenResponse{
-		Data: resources.Token{
-			Key: resources.Key{
-				ID:   req.Data.ID,
-				Type: resources.TOKEN,
-			},
-			Attributes: resources.TokenAttributes{
-				AccessToken: resources.Jwt{
-					Token:     access,
-					TokenType: string(jwt.AccessTokenType),
-				},
-				RefreshToken: resources.Jwt{
-					Token:     refresh,
-					TokenType: string(jwt.RefreshTokenType),
-				},
-			},
-		},
+	access, _, err := JWT(r).IssueJWT(&jwt.AuthClaim{
+		Nullifier: req.Data.ID,
+		Type:      jwt.AccessTokenType,
+	})
+	if err != nil {
+		return false, fmt.Errorf("failed to issueJWT: %w", err)
 	}
 
-	Cookies(r).SetAccessToken(w, access, aexp)
-	Cookies(r).SetRefreshToken(w, refresh, rexp)
-	ape.Render(w, resp)
+	cookies, err := cookiejar.New(nil)
+	if err != nil {
+		return false, fmt.Errorf("failed to create cookiejar: %w", err)
+	}
+
+	client := &http.Client{
+		Jar:     cookies,
+		Timeout: 5 * time.Second,
+	}
+
+	client.Jar.SetCookies(reqV.URL, []*http.Cookie{{
+		Name:  jwt.AccessTokenType.String(),
+		Value: access,
+	}})
+
+	resp, err := client.Do(reqV)
+	if err != nil {
+		return false, fmt.Errorf("failed to perform request: %w", err)
+	}
+	defer func() { resp.Body.Close() }()
+
+	respBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return false, fmt.Errorf("failed to read resp body: %w", err)
+	}
+
+	if resp.StatusCode != http.StatusOK {
```

Comment:
Also handle 404 and return 404 in handler in such case

Reply:
This is not necessary, because even if the user has no balance, he should still be able to receive the JWT